### PR TITLE
[BRA-1834] feat: Add hooks.metadata and deprecate hooks.meta

### PR DESCRIPTION
**Pull request by ibolmo**
## Changes
- Introduces new `hooks.metadata` (a reference to the metadata for the eval)
- Deprecates existing `hooks.meta` in favor of metadata
- Improves existing framework tests

## Why
This change provides a way for developers to read metadata and allows them a chance to mutate it without calling a function. 

## Testing
- Updated framework tests to verify new metadata functionality
- Verified backward compatibility with existing `hooks.meta` usage

### CODE REVIEW

File: js/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "braintrust",
-  "version": "0.0.160",
+  "version": "0.0.161",
```
Comment (ankrgyl) [inline js/package.json:3]: we update this via `version.py` when we bump the versions themselves. no need to here

**Review by ankrgyl — COMMENTED** [review]
Code lgtm. Can you add the equivalent in python too?

File: js/package.json (hunk 1 above)
Comment (ibolmo) [reply js/package.json]: gotcha. reverted.

**Review by ankrgyl — APPROVED** [review]

**Comment by ankrgyl** [comment]
Oh btw the failing test is b/c the type annotation is probably not supported in python3.8, which we still need to support for a wee bit longer

**Pull request merged**